release/models/acl/openconfig-packet-match.yang

```diff
@@ -394,12 +394,32 @@ module openconfig-packet-match {
         "Destination port or range";
     }
 
-    leaf-list tcp-flags {
-      type identityref {
-        base oc-pkt-match-types:TCP_FLAGS;
+    container transport-details {
+      choice explicit-or-builtin {
```

Comment:
We explicitly do not use the `choice`/`case` pattern in OpenConfig models.

We would rather do something like:

```yang

container transport-details {
   container config {
      description "...";

      leaf mode {
         type enumeration {
             enum BUILTIN { description "..."; };
             enum EXPLICIT { description "..."; };
         }
         description "...";
      }

      leaf-list explicit {
        type identityref { base oc-pkt-match-types:TCP_FLAGS; 
        when "../mode = 'EXPLICIT';
      }

      leaf explicit-match-option {
        type enumeration {
           enum ANY { description "..."; }
           enum ALL { description "..."; }
        }
        description "...";
      }

      leaf builtin {
        type enumeration {
          enum TCP_INITIAL { description "..."; }
          enum FRAGMENT { description "..."; }
          enum TCP_FRAGMENT { description "..."; }
        }
        description "...";
      }
   }
}
```